variables.tf

```diff
@@ -226,12 +226,34 @@ variable "tags" {
   description = "(Optional) Tags of the resource."
 }
 
-variable "key_vault_id" {
+variable "shared_subnet_id" {
   type        = string
```

Comment:
YOu should use the same approach here that you used for keyvault and storage account. to pass the resurce in.